packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -70,14 +72,47 @@ export function Comments( {
 		setIsConfirmDialogOpen( false );
 	};
 
-	const blockCommentId = useSelect( ( select ) => {
+	useSelect( ( select ) => {
 		const clientID = select( blockEditorStore ).getSelectedBlockClientId();
-		return (
+		const clientBlocks = select( blockEditorStore ).getBlocks();
+		setBlocksList( clientBlocks );
+
+		const getBlockCommentId =
 			select( blockEditorStore ).getBlock( clientID )?.attributes
-				?.blockCommentId ?? false
-		);
+				?.blockCommentId ?? false;
+
+		if ( getBlockCommentId ) {
+			setActiveClientId( getBlockCommentId );
+		}
 	}, [] );
 
+	const findBlockByCommentId = ( blocks, commentId ) => {
+		for ( const block of blocks ) {
+			if ( block.attributes.blockCommentId === commentId ) {
+				return block;
+			}
+			if ( block.innerBlocks && block.innerBlocks.length > 0 ) {
+				const foundBlock = findBlockByCommentId(
+					block.innerBlocks,
+					commentId
+				);
+				if ( foundBlock ) {
+					return foundBlock;
+				}
+			}
+		}
+		return null;
+	};
+
+	const handleThreadClick = ( thread ) => {
+		const block = findBlockByCommentId( blocksList, thread.id );
+		if ( block ) {
+			const blockClientId = block.clientId;
+			// Select the block in the editor
+			dispatch( blockEditorStore ).selectBlock( blockClientId );
```

Comment:
Please use `useDispatch` instead, so we get the actions from the correct registry based on the context

Reply:
@ellatrix - Updated with `useDispatch` as per your advice.